pkg/client/fake/client.go

```diff
@@ -85,6 +85,9 @@ func (t versionedTracker) Create(gvr schema.GroupVersionResource, obj runtime.Ob
 	if err != nil {
 		return err
 	}
+	if accessor.GetName() == "" {
```

Comment:
This needs to be a `NewInvalid`, this is how that looks like from a real api server:
```
              ErrStatus: {
                  TypeMeta: {Kind: "", APIVersion: ""},
                  ListMeta: {
                      SelfLink: "",
                      ResourceVersion: "",
                      Continue: "",
                      RemainingItemCount: nil,
                  },
                  Status: "Failure",
                  Message: "ConfigMap \"\" is invalid: metadata.name: Required value: name or generateName is required",
                  Reason: "Invalid",
                  Details: {
                      Name: "",
                      Group: "",
                      Kind: "ConfigMap",
                      UID: "",
                      Causes: [
                          {
                              Type: "FieldValueRequired",
                              Message: "Required value: name or generateName is required",
                              Field: "metadata.name",
                          },
                      ],
                      RetryAfterSeconds: 0,
                  },
                  Code: 422,
              }
```